Toolchain/CMake/Scripts/GeneratePluginTargetsAuto.cmake

```diff
@@ -0,0 +1,203 @@
+
+    # Get array length
+    string(JSON _len LENGTH "${_json_content}" custom_types)
+
+    set(_result "")
+    math(EXPR _last "${_len} - 1")
+
+    foreach(i RANGE 0 ${_last})
+        string(JSON _value GET "${_json_content}" custom_types ${i})
+        list(APPEND _result "${_value}")
+    endforeach()
+
+    set(${OUT_LIST} "${_result}" PARENT_SCOPE)
+endfunction()
+
+function(_nos_generate_plugin_target nos_plugin_file_path common_deps out_target_name)
+	message(STATUS "Configuring module ${plugin_name}")
+	nos_get_module_info_by_path(${nos_plugin_file_path} plugin_name plugin_version out_json_info)
+	
+	nos_normalize_plugin_name(${plugin_name} target_name)
+	get_filename_component(PLUGIN_DIR "${nos_plugin_file_path}" DIRECTORY)
+
+	nos_find_plugin_sdk_dependency(${out_json_info} plugin_sdk_version)
+	if ("${plugin_sdk_version}" STREQUAL "None")
+		message(STATUS "Module is not depended on a SDK version, there is no need for a target")
+		return()
+	endif()
+
+	message(STATUS "Plugin SDK version requested: ${plugin_sdk_version}")
+
+	add_compile_definitions(NOS_DISABLE_DEPRECATED)
+	nos_find_plugin_sdk(${plugin_sdk_version} NOS_PLUGIN_SDK_TARGET NOS_SDK_DIR)
+    if (NOT DEFINED NOS_SDK_DIR)
+        message(FATAL_ERROR "Nodos SDK with version ${plugin_sdk_version} not found, please either install it or choose different version")
+    endif()
+
+	nos_find_all_plugin_dependencies(${out_json_info} found_dependency_targets found_dep_dirs found_include_dirs)
+    list(APPEND INCLUDE_FOLDERS ${CMAKE_CURRENT_SOURCE_DIR} "${CMAKE_CURRENT_SOURCE_DIR}/Include" "${found_include_dirs}")
+    list(APPEND MODULE_DEPENDENCIES_TARGETS ${NOS_PLUGIN_SDK_TARGET})
+
+	_nos_get_custom_types_from_json(${nos_plugin_file_path} TYPE_FOLDERS)
+	if(TYPE_FOLDERS)
+    	nos_generate_flatbuffers("${TYPE_FOLDERS}" "${CMAKE_CURRENT_SOURCE_DIR}/Include/${target_name}" "cpp" "${NOS_SDK_DIR}/Types;${found_dep_dirs}" ${target_name}_generated)
+    	list(APPEND MODULE_DEPENDENCIES_TARGETS ${target_name}_generated)
+	endif()
+
+	# Check source code existence
+	if(NOT EXISTS "${PLUGIN_DIR}/Source")
+		message("Plugin has no source folder, no target generated")
+		return()
+	endif()
+
+	list(APPEND
+		MODULE_DEPENDENCIES_TARGETS
+    	${found_dependency_targets}
+    	${${common_deps}}   # dereference here
+	)
+	message(STATUS "Module dependencies: ${MODULE_DEPENDENCIES_TARGETS}")
+    nos_add_plugin("${target_name}" "${MODULE_DEPENDENCIES_TARGETS}" "${INCLUDE_FOLDERS}")
+	if(TARGET "${target_name}")
+		message(STATUS "Successfully created target: ${target_name}")
+	else()
+		nos_fatal_error("Failed to create target: ${target_name}")
+	endif()
+
+    #Helpers need C++20
+    set_target_properties("${target_name}" PROPERTIES CXX_STANDARD 20)
+	set(${out_target_name} ${target_name} PARENT_SCOPE)
+endfunction()
+
+function(_nos_process_plugin dir common_dependencies)
+	if(NOT IS_DIRECTORY ${dir})
+		nos_colored_message(COLOR RED "Can't process plugin because it's not folder")
```

Comment:
Error messages should contain reasons and info, in this case the path to the dir.